resources/assets/js/hooks/applicationHooks.tsx

```diff
@@ -271,6 +291,29 @@ export function useFetchApplication(
   return application;
 }
 
+/**
+ * Return an Application from the redux store, and fetch it from backend if it is not yet in the store.
+ * @param applicationId
+ * @param dispatch
+ */
+export function useFetchApplication(
+  applicationId: number,
+  dispatch: DispatchType,
+): Application | null {
+  const applicationSelector = (state: RootState): Application | null =>
+    getApplicationById(state, { id: applicationId });
+  const application: Application | null = useSelector(applicationSelector);
+  const applicationIsUpdating = useSelector((state: RootState) =>
+    getApplicationIsUpdating(state, { applicationId }),
+  );
+  useEffect(() => {
+    if (application === null && !applicationIsUpdating) {
+      dispatch(fetchApplication(applicationId));
+    }
+  }, [application, applicationId, applicationIsUpdating, dispatch]);
+  return application;
+}
```

Comment:
Okay, we need to be careful here. If the redux store has already fetched the Normalized Application for any reason, applicationSelector will be non null, but the application_review field will be undefined. You can see this in `constructNonNormalizedApplication` in applicationSelector.ts.
This might be unlikely to happen, where we fetch the Normalized Application and then later want the Reviewed Application, but we should be careful about avoiding it anyway. In the useEffect, we could check `(application === null || application.application_review === undefined)`.